Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -498,8 +498,13 @@ TextureCache::TCacheEntryBase* TextureCache::Load(const u32 stage)
 			}
 		}
 
-		// Find the entry which hasn't been used for the longest time
-		if (entry->frameCount != FRAMECOUNT_INVALID && entry->frameCount < temp_frameCount)
+		// Find the entry which hasn't been used for the longest time. Some games create
+		// animations by only changing the palette. Keeping all these textures in the cache
+		// improves the performance a lot for these games, so ignore paletted textures with
+		// the same base texture here. Example: Sonic the Fighters (inside Sonic Gems
+		// Collection) loops a 64 frames animation, where only the palette changes
+		if (entry->frameCount != FRAMECOUNT_INVALID && entry->frameCount < temp_frameCount &&
+			!(isPaletteTexture && entry->base_hash == tex_hash))
```

Comment:
I don't understand. If there are 2 textures in the same frame, both will have the frameCount FRAMECOUNT_INVALID, so neither will be invalidated.

And it's not fine to only increase TEXTURE_KILL_THRESHOLD, because that's only half of the problem. If the threshold is not big enough OR paletted textures are invalidated, then the game will load at least one texture per frame. With both changes, it's 0, at least with efb to texture. Efb to ram is a different story, if i remember correctly, but this pr is about fixing the regression i introduced.